### PR TITLE
feat: instance total public apps count

### LINKED ISSUE #27184: [Task]: analytics event for number of public apps in an instance
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Add a property to Instance stats which reports the number of public apps in the instance

### PR DESCRIPTION
**Pull request by NilanshBansal**
#### PR fixes following issue(s)
Fixes #27184

#### Type of change

- New feature (non-breaking change which adds functionality)
## Testing

- [x] Manual
- [x] Segment

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog1.java [hunk 1]
```diff
@@ -1600,40 +1600,32 @@ public void updateDatabaseDocumentationLinks_v1_2_1(MongoTemplate mongoTemplate)
         for (Plugin plugin : mongoTemplate.findAll(Plugin.class)) {
             switch (plugin.getPackageName()) {
                 case "postgres-plugin":
-                    plugin.setDocumentationLink(
-                            "https://docs.appsmith.com/reference/datasources/querying-postgres");
+                    plugin.setDocumentationLink("https://docs.appsmith.com/reference/datasources/querying-postgres");
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog1.java:1603]: Are these formatting changes due to spotless?

Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog1.java:1603]: Yes @sondermanish, I tried to ignore these but on running `mvn spotless:apply`, these re-appear

**Review by sondermanish — APPROVED** [review]
LGTM

**Pull request merged**